### ISSUE TITLE
🐛 The hash for development pods changes when they are moved to a new location

### ISSUE THREAD

**Issue by cragod** [swiftyfinch/Rugby#279]
## Did you already try Rugby doctor?
✅

## Describe the bug
I've identified an issue with development pods in our project. The hashing process is influenced by inconsistent paths(full path) across various development devices, leading to different hash values. Consequently, the cache cannot be shared with others, even for unmodified development pods.

## To Reproduce
Analyze the same development pods in different devices (or in the same device with different development pod paths).

## Expected behavior
I believe development pods should have the same hash value when their build contexts are identical.

## Screenshots
<img width="1091" alt="image" src="https://github.com/swiftyfinch/Rugby/assets/4200196/f87f5375-1aad-4dbb-9ae7-8b7118f113ca">
======
Update: After checking the repository in `/tmp`, I noticed that even some non-development pods also exhibit different path styles:

![image](https://github.com/swiftyfinch/Rugby/assets/4200196/88f3ad3c-0598-4de9-8b2b-5da9c32104b3)

## Environment (please complete the following information):
Version: 2.4.0

## Additional context
nil

**Comment by cragod** [comment]
I have an idea to execute the following tasks during the `targetHashContext` collection phase:
- Delete `PODS_TARGET_SRCROOT` if it exists.
- For all files' paths, remove the prefix corresponding to `$(PODS_TARGET_SRCROOT)`.

But I'm not sure if there are any overlooked details, I will give it a try.

**Comment by swiftyfinch** [comment]
@cragod Hello! Could you look at this related issues and pr? Maybe you will have some thoughts too.
- #256 
- #258

As I understand, at the first screenshot files placed outside of `Pods` directory. That's a problem for the current version of Rugby.
Yeah, and at the second one files placed outside of `$HOME` directory. I think Rugby cuts paths related to `$HOME`.

I need to think) I'll check your request 👍

**Comment by swiftyfinch** [comment]
@cragod Could you describe your project directory structure?
I can't understand what's wrong with `/Users/yunl/Workspace/RC/hello-mono-repo/dependencies/`.

I've tested and right now Rugby:
- Don't work with `/tmp` directory because there is a symlink. I can fix it;
- Works with basic project directory structure:
  ```sh
  Example
  ├─ Pods
  └─ LocalPods
  ```
- Works with `React Native` project directory structure:
  ```sh
  RAProject
  └─ ios
     └─ Pods
  ```

So, I need to understand which paths saved as absolute ones in your hash yml files?
Also, I need to know an absolute path to your `Pods` directory.

And finally, you can debug this place by yourself:
https://github.com/swiftyfinch/Rugby/blob/4435df64b03a41233af53f2a83a0c93371e89b7d/Sources/RugbyFoundation/Core/Common/Hashers/FileContentHasher.swift#L42

**Comment by cragod** [comment]
There are several modules under `dependencies`, something like:
```
.
- app
  - ios
    - ...
    - Podfile
  - android
- dependencies
  - moduleA
    - ...
    - moduleA.podspec
  - module B
    - ...
    - moduleB.podspec
```

Podfile:
```
pod 'moduleA', :path => '../../dependencies/moduleA'
pod 'moduleB', :path => '../../dependencies/moduleB'
```

Then, in my practice, the source files of moduleA and moduleB are resolved to the full local path in `targetHashContext`.

**Comment by swiftyfinch** [comment]
@cragod Could you test #288?
It works good to me ✅.

**Comment by cragod** [comment]
> @cragod Could you test #288? It works good to me ✅.

I tried it, and it works well. The paths in the hash YAML are no longer linked to the repo location 👍🏻

**Comment by swiftyfinch** [comment]
@cragod Great, I've merged #288. 👍 Can we close this issue and your draft?
I can release a patch right after we will finish with this issue.
